editor/shader/shader_create_dialog.cpp

```diff
@@ -408,7 +316,43 @@ void ShaderCreateDialog::_path_submitted(const String &p_path) {
 	}
 }
 
-void ShaderCreateDialog::config(const String &p_base_path, bool p_built_in_enabled, bool p_load_enabled, int p_preferred_type, int p_preferred_mode) {
+void ShaderCreateDialog::config(const String &p_base_path, bool p_built_in_enabled, bool p_load_enabled, const String &p_preferred_type, int p_preferred_mode) {
+	_update_language_info();
+	type_menu->clear();
+	const Vector<Ref<EditorShaderLanguagePlugin>> shader_langs = EditorShaderLanguagePlugin::get_shader_languages_read_only();
+	ERR_FAIL_COND_MSG(shader_langs.is_empty(), "ShaderCreateDialog: Unable to load any shader languages!");
+	for (Ref<EditorShaderLanguagePlugin> shader_lang : shader_langs) {
+		const PackedStringArray variations = shader_lang->get_language_variations();
+		for (const String &variation : variations) {
+			type_menu->add_item(variation);
+		}
+	}
+	_refresh_type_icons();
```

Comment:
If you are refreshing the icons here, you can remove the call from `NOTIFICATION_THEME_CHANGED`.